.hooks/pre-commit.sh

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```

Comment:
Why that script and not regular pre-commit configuration file?

Reply:
Does that mean the user needs https://pre-commit.com/ ?

I've had some... trouble... convincing fellow developers to use pre-commit mechanisms.

We do, however, need the same style/lint CI as nvim-tree. Future PR...

Reply:
If contributors don't have pre-commit set up it will be a noop, so no real downside I think. We can then use pre-commit on CI to run the same checks that local pre-commit would do and fail if any files would be changed by set up tasks, which would then fallback to contributor to fix those issues.

Reply:
> If contributors don't have pre-commit set up it will be a noop

Sounds good to me. It's really up to them to set things up locally.

> CI to run the same checks that local pre-commit would do

That works beautifully for nvim-tree.

Can I please ask you to make the changes here? We can set up CI as soon as this is merged.

Reply:
@alex-courtis please see #233

Reply:
Many thanks!